go-controller/pkg/cni/ovs.go

```diff
@@ -98,17 +99,29 @@ func ofctlExec(args ...string) (string, error) {
 	return stdoutStr, nil
 }
 
-func waitForPodFlows(mac string) error {
+func waitForPodFlows(mac string, ipNets []*net.IPNet) error {
 	return wait.PollImmediate(200*time.Millisecond, 20*time.Second, func() (bool, error) {
-		// Query the flows by mac address
-		query := fmt.Sprintf("table=9,dl_src=%s", mac)
-		// ovs-ofctl dumps error on stderr, so stdout will only dump flow data if matches the query.
-		stdout, err := ofctlExec("dump-flows", "br-int", query)
-		if err != nil {
-			return false, nil
+		// Query the flows by mac address for in_port_security
+		queries := []string{fmt.Sprintf("table=9,dl_src=%s", mac)}
+		for _, ipNet := range ipNets {
+			var ipMatch string
+			if ipNet.IP.To4() != nil {
+				ipMatch = "ip_dst"
+			} else {
+				ipMatch = "ip6_dst"
+			}
+			// add queries for out_port_security
+			queries = append(queries, fmt.Sprintf("table=48,ip,%s=%s", ipMatch, ipNet.IP))
 		}
-		if len(stdout) == 0 {
-			return false, nil
+		for _, query := range queries {
+			// ovs-ofctl dumps error on stderr, so stdout will only dump flow data if matches the query.
+			stdout, err := ofctlExec("dump-flows", "br-int", query)
```

Comment:
I am not sure batching the query will work here with ovs-ofctl? But even if it works, it can give you false positive, in a scenario where only one table has a flow but second table doesn't. Stdout will still have some flow as a output, but that doesn't necessary mean that both the tables has the required flow. So seems like you need to separate the query here ?

Reply:
If the 2nd iteration didn't have an entry or failed, then stdout would still be set to the output of that command. If either of those checks fail for any query, it will return false and the function will start over until all queries are successful.